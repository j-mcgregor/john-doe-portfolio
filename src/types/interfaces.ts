export interface NavbarProps {
    brand: string | JSX.Element
}

export interface PRISMIC_RichTextType {
    type: string
    text: string
    spans: any[]
}

export interface PRISMIC_Image {
    dimensions: {
        width: number
        height: number
    }
    alt?: string | null
    copyright: string | null
    url: string
}

export interface PRISMIC_ExperienceType {
    job_title: string
    date_from: string
    date_to?: string
    present?: boolean
    company: string
    city: string
    description: PRISMIC_RichTextType[]
}

export interface PRISMIC_landing_node {
    node: {
        primary_text: PRISMIC_RichTextType[]
        secondary_text: PRISMIC_RichTextType[]
        background_image: PRISMIC_Image
        about: PRISMIC_RichTextType[]
        experience: PRISMIC_ExperienceType[]
    }
}

export interface IndexPageProps {
    data: {
        prismic: {
            allLandings: {
                edges: PRISMIC_landing_node[]
            }
        }
        file: {
            childImageSharp: {
                fixed: {
                    height: number
                    width: number
                    base64: string
                    src: string
                    srcSet: string
                }
            }
        }
    }
}

export interface HeroProps {
    primary: JSX.Element
    secondary: JSX.Element
    logo?: JSX.Element
    backgroundImage?: string
}

export interface StyledHeroProps {
    bgImageUrl?: string
}

export interface AboutProps {
    about: JSX.Element
    experience: JSX.Element[]
}

// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  FORMS   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

export interface TextInputProps {
    type: string
    placeholder: string
    label?: string
    name: string
    required: boolean
    classNames?: string[]
    value: string
    onChange: (e: React.FormEvent<HTMLInputElement>) => void
}

export interface TextAreaInputProps {
    placeholder: string
    label?: string
    name: string
    required: boolean
    value: string
    onChange: (e: React.FormEvent<HTMLTextAreaElement>) => void
}
