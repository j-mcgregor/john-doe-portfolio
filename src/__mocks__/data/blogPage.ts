import _ from 'lodash'
import {
    BlogPageProps,
    PRISMIC_PRIMARY_Image_with_Caption,
    PRISMIC_PRIMARY_Quote,
    PRISMIC_PRIMARY_Text,
} from '../../types/interfaces/blog'
import {
    PRISMIC_blog_node,
    PRISMIC_RichTextType,
    PRISMIC_MetaType,
} from '../../types/interfaces/prismic'

type ImageType = 'image_with_caption' | 'full-width-image' | 'emphasized'

type TextType = 'paragraph' | 'heading3'

export interface ImageSliceTypeProps {
    [index: string]: ImageType
}

export const imageSliceType: ImageSliceTypeProps = {
    '0': 'image_with_caption',
    '1': 'full-width-image',
    '2': 'emphasized',
}

export const makeTextType = (
    type: TextType,
    text: string
): PRISMIC_RichTextType => ({
    type,
    text,
    spans: [],
})

export const meta = (
    id: string,
    uid: string,
    type: string
): PRISMIC_MetaType => ({
    id,
    uid,
    type,
})

export const imageSlice = (
    imageType: number
): PRISMIC_PRIMARY_Image_with_Caption => ({
    primary: {
        caption: [makeTextType('paragraph', `caption-${imageType}`)],
        image: {
            dimensions: {
                width: 1200,
                height: 900,
            },
            alt: `alt-${imageType}`,
            copyright: null,
            url: 'http://fillmurray.com/300/300',
        },
    },
    label: `default`,
    type: imageSliceType[`${imageType}`],
})

export const quoteSlice = (text: string): PRISMIC_PRIMARY_Quote => ({
    primary: {
        quote: [makeTextType('paragraph', text)],
    },
    label: undefined,
    type: 'quote',
})

export const textSlice = (text: string): PRISMIC_PRIMARY_Text => ({
    primary: {
        text: [makeTextType('paragraph', text)],
    },
    label: undefined,
    type: 'text',
})

export const blogNodes: PRISMIC_blog_node[] = _.times(3, num => ({
    node: {
        date: `2020-0${num + 1}-01`,
        body: [
            textSlice('Lorem ipsum dolor sit amet'),
            imageSlice(num),
            quoteSlice('And I quoteth thus...'),
        ],
        _meta: meta(`id-${num + 1}`, `blog-${num + 1}`, 'blog_post'),
        title: [makeTextType('paragraph', `Post ${num + 1}`)],
    },
}))

export const blogPage: BlogPageProps = {
    data: {
        prismic: {
            allBlog_posts: {
                edges: blogNodes,
            },
        },
    },
}
