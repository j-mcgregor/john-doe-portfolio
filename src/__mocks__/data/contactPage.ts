export const contactPage = {
    data: {
        prismic: {
            allContacts: {
                edges: [
                    {
                        node: {
                            title: [
                                {
                                    type: 'heading3',
                                    text: 'Contact Us',
                                    spans: [],
                                },
                            ],
                            subtitle: [
                                {
                                    type: 'heading4',
                                    text: 'Leave a message',
                                    spans: [],
                                },
                            ],
                            social_links: [],
                        },
                    },
                ],
            },
        },
    },
}
