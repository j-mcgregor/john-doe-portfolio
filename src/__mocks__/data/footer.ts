export const footerData = {
    prismic: {
        allContacts: {
            edges: [
                {
                    node: {
                        social_links: [
                            {
                                name: 'Mail',
                                text: 'info@johndoe.com',
                                url: {
                                    url: 'mailto:info@johndoe.com',
                                },
                            },
                            {
                                name: 'Phone',
                                text: '+12345678900',
                                url: {
                                    url: 'tel:+12345678900',
                                },
                            },
                            {
                                name: 'LinkedIn',
                                text: null,
                                url: {
                                    url: 'https://linkedin.com',
                                },
                            },
                            {
                                name: 'Facebook',
                                text: null,
                                url: {
                                    url: 'https://facebook.com',
                                },
                            },
                            {
                                name: 'Twitter',
                                text: null,
                                url: {
                                    url: 'https://twitter.com',
                                },
                            },
                            {
                                name: 'Address',
                                text: '123 Main Street, New York, USA',
                                url: null,
                            },
                        ],
                    },
                },
            ],
        },
    },
}
