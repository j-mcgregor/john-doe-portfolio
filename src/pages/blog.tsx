import React from 'react'
import { graphql } from 'gatsby'

import Layout from '../components/Layout'
import SEO from '../components/SEO'
import { BlogPageProps } from '../types/interfaces/blog'
import BlogPosts from '../components/blog/BlogPosts'

export const query = graphql`
    query BlogPostsQuery {
        prismic {
            allBlog_posts {
                edges {
                    node {
                        title
                        date
                        body {
                            ... on PRISMIC_Blog_postBodyImage_with_caption {
                                primary {
                                    caption
                                    image
                                }
                                type
                            }
                            ... on PRISMIC_Blog_postBodyText {
                                primary {
                                    text
                                }
                                type
                            }
                        }
                        _meta {
                            id
                            type
                            uid
                        }
                    }
                }
            }
        }
    }
`

const BlogPage: React.FC<BlogPageProps> = ({
    data: {
        prismic: {
            allBlog_posts: { edges },
        },
    },
}) => {
    return (
        <Layout>
            <SEO title="Blog Post" />
            <BlogPosts posts={edges} />
        </Layout>
    )
}

export default BlogPage
