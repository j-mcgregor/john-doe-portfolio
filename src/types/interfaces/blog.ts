// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>  BLOG   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import {
    PRISMIC_RichTextType,
    PRISMIC_Image,
    PRISMIC_blog_node,
    PRISMIC_blog_node_body,
} from './prismic'

export interface PRISMIC_PRIMARY_Image_with_Caption {
    type: string
    label?: string
    primary?: {
        caption?: PRISMIC_RichTextType[]
        image?: PRISMIC_Image
    }
}

export interface PRISMIC_PRIMARY_Text {
    type: string
    label?: string
    primary?: {
        text?: PRISMIC_RichTextType[]
    }
}

export interface PRISMIC_PRIMARY_Quote {
    type: string
    label?: string
    primary?: {
        quote?: PRISMIC_RichTextType[]
    }
}

export interface BlogPageProps {
    data: {
        prismic: {
            allBlog_posts: {
                edges: PRISMIC_blog_node[]
            }
        }
    }
}

export interface BlogBodyProps {
    body: Array<
        | PRISMIC_PRIMARY_Image_with_Caption
        | PRISMIC_PRIMARY_Quote
        | PRISMIC_PRIMARY_Text
    >
}

export interface PostSummaryProps {
    post: PRISMIC_blog_node_body
}

export interface BlogPostsProps {
    posts?: PRISMIC_blog_node[]
}

export interface ImageCaptionProps {
    slice: PRISMIC_PRIMARY_Image_with_Caption
}
