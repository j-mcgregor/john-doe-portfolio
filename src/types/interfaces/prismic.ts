// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  PRISMIC   >>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import { BlogBodyProps } from './blog'

export interface PRISMIC_RichTextType {
    spans: any[]
    text: string
    type: string
}

export interface PRISMIC_Image {
    alt?: string | null
    copyright: string | null
    dimensions: {
        height: number
        width: number
    }
    url: string
    thumbnail?: {
        dimensions: {
            width: number
            height: number
        }
        alt: string | null
        copyright: string | null
        url: string
    }
}

export interface PRISMIC_ExperienceType {
    city: string
    company: string
    date_from: string
    date_to?: string
    description: PRISMIC_RichTextType[]
    job_title: string
    present?: boolean
}

export interface PRISMIC_SocialLinks {
    name: string
    text?: string | null
    url?: {
        url: string
    } | null
}

export interface PRISMIC_gallery_body {
    type: string
    label?: string
    fields: PRISMIC_gallery_image_fields[]
}

export interface PRISMIC_gallery_image_fields {
    alt_text: PRISMIC_RichTextType[] | null
    image: PRISMIC_Image
    caption: PRISMIC_RichTextType[] | null
}

// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  NODES   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

export interface PRISMIC_landing_node {
    node: {
        about: PRISMIC_RichTextType[]
        background_image: PRISMIC_Image
        experience: PRISMIC_ExperienceType[]
        primary_text: PRISMIC_RichTextType[]
        secondary_text: PRISMIC_RichTextType[]
    }
}

export interface PRISMIC_contact_node {
    node: {
        title: PRISMIC_RichTextType[] | null
        subtitle: PRISMIC_RichTextType[] | null
        social_links: PRISMIC_SocialLinks[]
    }
}

export interface PRISMIC_gallery_node {
    node: {
        title: PRISMIC_RichTextType[] | null
        description: PRISMIC_RichTextType[] | null
        body: PRISMIC_gallery_body[]
    }
}

export interface PRISMIC_blog_node_body {
    title: PRISMIC_RichTextType[]
    date: string
    body: BlogBodyProps[]
    _meta: {
        id: string
        uid: string
        type: string
    }
}
export interface PRISMIC_blog_node {
    node: PRISMIC_blog_node_body
}
