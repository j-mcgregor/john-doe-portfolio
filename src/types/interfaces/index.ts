// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>/  INDEX   >>>>>>>>>>>>>>>>>>>>
// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

import { PRISMIC_landing_node } from './prismic'

export interface IndexPageProps {
    data: {
        prismic: {
            allLandings: {
                edges: PRISMIC_landing_node[]
            }
        }
        file: {
            childImageSharp: {
                fixed: {
                    height: number
                    width: number
                    base64: string
                    src: string
                    srcSet: string
                }
            }
        }
    }
}
