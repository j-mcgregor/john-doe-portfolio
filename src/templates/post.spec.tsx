import React from 'react'
import { render } from '@testing-library/react'
import * as Gatsby from 'gatsby'
import PostTemplate, { PostBody } from './post'
import { BlogPageProps } from '../types/interfaces/blog'
import * as data from '../__mocks__/data/blogPage'
import { contactPage } from '../__mocks__/data/contactPage'
import { PRISMIC_blog_node_body } from '../types/interfaces/prismic'

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery')

describe('PostTemplate', () => {
    let props: BlogPageProps
    beforeEach(() => {
        // Component renders Layout which requires Contact data
        useStaticQuery.mockImplementation(() => ({
            ...contactPage.data,
        }))

        props = data.blogPage
    })

    it('should render with all information', () => {
        const { baseElement } = render(<PostTemplate {...props} />)

        expect(baseElement).toMatchSnapshot()
    })
})

describe('PostBody', () => {
    let props: PRISMIC_blog_node_body
    beforeEach(() => {
        // Component renders Layout which requires Contact data
        useStaticQuery.mockImplementation(() => ({
            ...contactPage.data,
        }))

        props = data.blogBody(0)
    })

    it('should render', () => {
        const { baseElement } = render(<PostBody {...props} />)

        expect(baseElement).toMatchSnapshot()
    })

    it('should render with no title', () => {
        const newProps: PRISMIC_blog_node_body = {
            ...props,
            title: [data.makeTextType('paragraph', '')],
        }

        const { baseElement } = render(<PostBody {...newProps} />)

        expect(baseElement).toMatchSnapshot()
    })
})
