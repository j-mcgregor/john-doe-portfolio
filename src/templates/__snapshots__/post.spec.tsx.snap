// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostBody should render 1`] = `
<body>
  <div>
    <div
      class="blog-page"
    >
      <div
        class="container post-header"
      >
        <div
          class="back"
        >
          <a
            href="/blog"
          >
            Back
          </a>
        </div>
        <h1
          data-wio-id="id-0"
        >
          Post 0
        </h1>
      </div>
      <div>
        <div
          class="homepage-slice-wrapper"
        >
          <div
            class="post-text container"
          >
            <div>
              <p>
                Lorem ipsum dolor sit amet
              </p>
            </div>
          </div>
        </div>
        <div
          class="homepage-slice-wrapper"
        >
          <div
            class="post-image container"
          >
            <figcaption
              class="block-img"
            >
              <img
                alt="alt-0"
                src="http://fillmurray.com/300/300"
              />
              <figcaption
                class="image-label"
              >
                caption-0
              </figcaption>
            </figcaption>
          </div>
        </div>
        <div
          class="homepage-slice-wrapper"
        >
          <div
            class="post-quote container"
          >
            <blockquote>
              <p>
                And I quoteth thus...
              </p>
            </blockquote>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`PostBody should render with no title 1`] = `
<body>
  <div>
    <div
      class="blog-page"
    >
      <div
        class="container post-header"
      >
        <div
          class="back"
        >
          <a
            href="/blog"
          >
            Back
          </a>
        </div>
        <h1
          data-wio-id="id-0"
        />
      </div>
      <div>
        <div
          class="homepage-slice-wrapper"
        >
          <div
            class="post-text container"
          >
            <div>
              <p>
                Lorem ipsum dolor sit amet
              </p>
            </div>
          </div>
        </div>
        <div
          class="homepage-slice-wrapper"
        >
          <div
            class="post-image container"
          >
            <figcaption
              class="block-img"
            >
              <img
                alt="alt-0"
                src="http://fillmurray.com/300/300"
              />
              <figcaption
                class="image-label"
              >
                caption-0
              </figcaption>
            </figcaption>
          </div>
        </div>
        <div
          class="homepage-slice-wrapper"
        >
          <div
            class="post-quote container"
          >
            <blockquote>
              <p>
                And I quoteth thus...
              </p>
            </blockquote>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`PostTemplate should render with all information 1`] = `
<body>
  <div>
    <nav
      class="navbar"
    >
      <a
        class="brand"
        href="/"
      >
        John Doe
      </a>
      <ul
        class="nav-list"
      >
        <li
          class="nav-item"
        >
          <a
            href="/gallery"
          >
            Gallery
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="container-fluid"
    >
      <div
        class="blog-page"
      >
        <div
          class="container post-header"
        >
          <div
            class="back"
          >
            <a
              href="/blog"
            >
              Back
            </a>
          </div>
          <h1
            data-wio-id="id-1"
          >
            Post 1
          </h1>
        </div>
        <div>
          <div
            class="homepage-slice-wrapper"
          >
            <div
              class="post-text container"
            >
              <div>
                <p>
                  Lorem ipsum dolor sit amet
                </p>
              </div>
            </div>
          </div>
          <div
            class="homepage-slice-wrapper"
          >
            <div
              class="post-image container"
            >
              <figcaption
                class="block-img"
              >
                <img
                  alt="alt-0"
                  src="http://fillmurray.com/300/300"
                />
                <figcaption
                  class="image-label"
                >
                  caption-0
                </figcaption>
              </figcaption>
            </div>
          </div>
          <div
            class="homepage-slice-wrapper"
          >
            <div
              class="post-quote container"
            >
              <blockquote>
                <p>
                  And I quoteth thus...
                </p>
              </blockquote>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="footer"
    >
      Reach me on
      <div
        class="social"
      />
    </footer>
  </div>
</body>
`;
